component/src/main/java/io/siddhi/extension/store/elasticsearch/sink/ElasticsearchSink.java

```diff
@@ -0,0 +1,287 @@
+        },
+        examples = {
+                @Example(syntax = "" +
+                        "@sink(type='elasticsearch', hostname='172.0.0.1', port='9200'," +
+                        "index.name='stock_index', " +
+                        "@map(type='json', @payload(\"\"\"{\n" +
+                        "   \"Stock Data\":{\n" +
+                        "      \"Symbol\":\"{{symbol}}\",\n" +
+                        "      \"Price\":{{price}},\n" +
+                        "      \"Volume\":{{volume}}\n" +
+                        "   }\n" +
+                        "}\"\"\")))" +
+                        "define stream stock_stream(symbol string, price float, volume long);",
+                        description = "This will create an index called 'stock_index' if it does not already exist " +
+                                "in the elasticsearch server and saves the custom json document."
+                )
+        }
+)
+
+public class ElasticsearchSink extends Sink {
+
+    private static final Logger logger = Logger.getLogger(ElasticsearchSink.class);
+    private ElasticsearchConfigs elasticsearchConfigs;
+
+    @Override
+    public Class[] getSupportedInputEventClasses() {
+        return new Class[]{String.class};
+    }
+
+    @Override
+    protected ServiceDeploymentInfo exposeServiceDeploymentInfo() {
+        return null;
+    }
+
+    @Override
+    public String[] getSupportedDynamicOptions() {
+        return new String[0];
+    }
+
+    @Override
+    protected StateFactory init(StreamDefinition streamDefinition, OptionHolder optionHolder,
+                                ConfigReader sinkConfigReader, SiddhiAppContext siddhiAppContext) {
+        ElasticsearchConfigs.setLogger(logger);
+        elasticsearchConfigs = new ElasticsearchConfigs(this);
+        elasticsearchConfigs.init(streamDefinition, sinkConfigReader, siddhiAppContext);
+        return null;
+    }
+
+    @Override
+    public void publish(Object payload, DynamicOptions dynamicOptions, State state)
+            throws ConnectionUnavailableException {
+        SiddhiIndexRequest siddhiIndexRequest = new SiddhiIndexRequest(payload, dynamicOptions,
+                elasticsearchConfigs.getIndexName());
+        try {
+            JsonElement parse = new JsonParser().parse((String) payload);
+            if (parse.isJsonArray()) {
+                JsonArray jsonArray = parse.getAsJsonArray();
+                for (JsonElement jsonElement : jsonArray) {
+                    siddhiIndexRequest.source(jsonElement.toString(), XContentType.JSON);
+                    elasticsearchConfigs.getBulkProcessor().add(siddhiIndexRequest);
+                }
+                logger.debug(payload + " has been successfully added.");
+            } else if (parse.isJsonObject()) {
+                siddhiIndexRequest.source(parse.getAsJsonObject().toString(), XContentType.JSON);
+                elasticsearchConfigs.getBulkProcessor().add(siddhiIndexRequest);
+            }
```

Comment:
Whether it could have other than json object or json array ?

Reply:
When I test the elasticsearch sink with the editor, I got the json string as json object, but when I run the test case in the project I got the json string as json array. That's why added support for both the json object and the json array.